lib/DebugInfo/Symbolize/BugsnagSymbolize.cpp

```diff
@@ -0,0 +1,100 @@
+#include "llvm/DebugInfo/Symbolize/BugsnagSymbolize.h"
+#include "llvm/DebugInfo/Symbolize/Symbolize.h"
+#include <cstdlib>
+#include <cstring>
+
+using namespace llvm;
+
+static void freeAndInvalidate(void** p) {
+  free((void*)*p);
+  *p = nullptr;
+}
+
+static void destroySymbolizeResult(SymbolizeResult* symbolizeResult) {
+  if (!symbolizeResult) {
+    return;
+  }
+  freeAndInvalidate((void**)&symbolizeResult->fileName);
+  freeAndInvalidate((void**)&symbolizeResult->shortFunctionName);
+  freeAndInvalidate((void**)&symbolizeResult->linkageFunctionName);
+  freeAndInvalidate((void**)&symbolizeResult->symbolTableFunctionName);
+}
+
+static void destroySymbolizeResults(SymbolizeResults* symbolizeResults) {
+  if (!symbolizeResults) {
+    return;
+  }
+  for (int i = 0; i < symbolizeResults->resultCount; i++) {
+    destroySymbolizeResult(&symbolizeResults->results[i]);
```

Comment:
The "results" field here is already a pointer, so this looks to me like you're creating a pointer to a pointer, when you just want the initial pointer.

Reply:
As discussed this can stay as is.